wagtail/admin/filters.py

```diff
@@ -105,16 +105,24 @@ def __init__(self, data=None, queryset=None, *, request=None, prefix=None):
     def _add_locale_filter(self):
         # Add a locale filter if the model is translatable
         # and there isn't one already.
-        from wagtail.models.i18n import TranslatableMixin
+        from wagtail.models.i18n import Locale, TranslatableMixin
 
         if (
             self._meta.model
             and issubclass(self._meta.model, TranslatableMixin)
             and "locale" not in self.filters
         ):
+            # Only add the locale filter if there are multiple content languages
+            # in the settings and the corresponding Locales exist.
+            languages = get_content_languages()
+            locales = set(Locale.objects.values_list("language_code", flat=True))
+            choices = [(k, v) for k, v in languages.items() if k in locales]
```

Comment:
Would be nice to have this as a util that is cached, similar to `get_locales_display_names` which uses `id` as the key. Or maybe we can combine the two be in a single cache key...

For now I opted for a simpler solution so this isn't very controversial to merge into 6.0 during feature freeze.